src/dyncfg/src/lib.rs

```diff
@@ -242,45 +229,123 @@ impl ConfigEntry {
         &self.default
     }
 
-    /// The sharable value of this config in the set.
-    pub fn val(&self) -> &ConfigVal {
-        &self.val
+    /// The value of this config in the set.
+    pub fn val(&self) -> ConfigVal {
+        self.val.load()
     }
 }
 
-/// A type-erased [ConfigType::Shared] for when set of different types are
-/// stored in a collection.
-///
-/// TODO(cfg): Consider moving these Arcs to be a single one around the map in
-/// `ConfigSet` instead. That would mean less pointer-chasing in the common
-/// case, but would remove the possibility of amortizing the name lookup via
-/// [Config::get_from_shared].
+/// A type-erased configuration value for when set of different types are stored
+/// in a collection.
 #[derive(Clone, Debug)]
 pub enum ConfigVal {
-    /// A `bool` shared value.
+    /// A `bool` value.
+    Bool(bool),
+    /// A `u32` value.
+    U32(u32),
+    /// A `usize` value.
+    Usize(usize),
+    /// An `Option<usize>` value.
+    OptUsize(Option<usize>),
+    /// A `String` value.
+    String(String),
+    /// A `Duration` value.
+    Duration(Duration),
+}
+
+/// An atomic version of [`ConfigVal`] to allow configuration values to be
+/// shared between configuration writers and readers.
+// TODO(cfg): Consider moving these Arcs to be a single one around the map in
```

Comment:
super nit: Preference for an empty comment line in between these two "paragraphs" please

Reply:
Done.

The reason I didn't have a paragraph break there is because the first paragraph is a documentation comment (///) and the second paragraph is not. Having a blank line in between is something I've never figured how to neatly do; it's always either a trailing empty line for the documentation comment or a leading empty line for the non-documentation comment. So I've settled on "no space" as the best of the bad options.

It looks like persist/dyncfg style is just to make the TODO comments part of the documentation comment to sidestep the question, though that has the alternative defect of making the implementation TODOs visible to the library user in the rendered documentation. :shrugging: Anyway, I've matched that style here and added the blank line.